components/project_scatter.py

```diff
@@ -1,78 +1,112 @@
+                                value='Bliss_excess',
+                                id='y-axis-select'
+                            )
+                        ]))]),
+                dbc.Col(
+                    width=4,
+                    className='mt-2 mb-4',
+                    children=[
+                        dbc.Form(inline=True, children=dbc.FormGroup([
+                            html.Label('Color'),
+                            dcc.Dropdown(
+                                options=[{'label': l, 'value': v} for l, v in
+                                         [('Default', 'default'),
+                                          ("Cell Line", 'model_name'),
+                                          ("Tissue", 'tissue'),
+                                          ("Combination", 'combo_id')]],
+                                value='default',
+                                id='color-select'
+                            )
+                        ]))]),
+             ]),
+            dbc.Row([
+                dbc.Col(
+                    width=6,
+                    className='mt-2 mb-4',
+                    children=[
+                        dbc.Form(inline=True, children=dbc.FormGroup([
+                            html.Label('Tissue'),
+                            dcc.Dropdown(
+                                options=[{'label': c, 'value': c} for c in tissues()],
+                                id='tissue-select',
+                                multi=True
+                            ),
+                        ]))]),
+                dbc.Col(
+                    width=6,
+                    className='mt-2 mb-4',
+                    children=[
+                        dbc.Form(inline=True, children=dbc.FormGroup([
+                            html.Label('Combination'),
+                            dcc.Dropdown(
+                                options=[{'label': f"{c.lib1.drug_name} + {c.lib2.drug_name}", 'value': f"{c.lib1_id}+{c.lib2_id}"} for c in project.combinations],
+                                id='combination-select',
+                                multi=True,
+                            ),
+                        ]))]),
+            ]),
+            dbc.Row(id='correlation', className='ml-3 mt-2'),
+            dbc.Row(dcc.Graph(id='project-scatter'))
+        ])
+    ])
 
 @app.callback(
     dash.dependencies.Output('project-scatter', 'figure'),
     [dash.dependencies.Input('x-axis-select', 'value'),
      dash.dependencies.Input('y-axis-select', 'value'),
      dash.dependencies.Input('color-select', 'value'),
+     dash.dependencies.Input('tissue-select', 'value'),
+     dash.dependencies.Input('combination-select', 'value'),
      dash.dependencies.Input('project-id', 'children')])
-def update_scatter(x_axis_field, y_axis_field, color_field, project_id):
+def update_scatter(x_axis_field, y_axis_field, color_field, tissues, combinations, project_id):
```

Comment:
The variable names should be more descriptive. Perhaps something like 'selected_tissues' and 'selected_combinations'

Reply:
I
```suggestion
def update_scatter(x_axis_field, y_axis_field, color_field, selected_tissues, selected_combinations, project_id):
```